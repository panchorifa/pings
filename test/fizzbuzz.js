const fizzbuzz = require('../src/fizzbuzz')

const MAPPING = {
  3: 'fizz',
  5: 'buzz',
  7: 'hello',
  11: 'goodbye'
}

describe('fizz', () => {
  test('buzz', () => {
    const msgs = fizzbuzz(21, MAPPING)
    expect(msgs[0]).toBe(1)
    expect(msgs[2]).toBe(MAPPING[3])
    expect(msgs[4]).toBe(MAPPING[5])
    expect(msgs[6]).toBe(MAPPING[7])
    expect(msgs[10]).toBe(MAPPING[11])
    expect(msgs[14]).toBe(`${MAPPING[3]}${MAPPING[5]}`)
    expect(msgs[15]).toBe(16)
    expect(msgs[20]).toBe(`${MAPPING[3]}${MAPPING[7]}`)
  })

  test('all factors', () => {
    expect(fizzbuzz(110, MAPPING)).toMatchSnapshot()
  })
})
