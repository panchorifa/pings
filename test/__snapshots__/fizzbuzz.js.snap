// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fizz all factors 1`] = `
Array [
  1,
  2,
  "fizz",
  4,
  "buzz",
  "fizz",
  "hello",
  8,
  "fizz",
  "buzz",
  "goodbye",
  "fizz",
  13,
  "hello",
  "fizzbuzz",
  16,
  17,
  "fizz",
  19,
  "buzz",
  "fizzhello",
  "goodbye",
  23,
  "fizz",
  "buzz",
  26,
  "fizz",
  "hello",
  29,
  "fizzbuzz",
  31,
  32,
  "fizzgoodbye",
  34,
  "buzzhello",
  "fizz",
  37,
  38,
  "fizz",
  "buzz",
  41,
  "fizzhello",
  43,
  "goodbye",
  "fizzbuzz",
  46,
  47,
  "fizz",
  "hello",
  "buzz",
  "fizz",
  52,
  53,
  "fizz",
  "buzzgoodbye",
  "hello",
  "fizz",
  58,
  59,
  "fizzbuzz",
  61,
  62,
  "fizzhello",
  64,
  "buzz",
  "fizzgoodbye",
  67,
  68,
  "fizz",
  "buzzhello",
  71,
  "fizz",
  73,
  74,
  "fizzbuzz",
  76,
  "hellogoodbye",
  "fizz",
  79,
  "buzz",
  "fizz",
  82,
  83,
  "fizzhello",
  "buzz",
  86,
  "fizz",
  "goodbye",
  89,
  "fizzbuzz",
  "hello",
  92,
  "fizz",
  94,
  "buzz",
  "fizz",
  97,
  "hello",
  "fizzgoodbye",
  "buzz",
  101,
  "fizz",
  103,
  104,
  "fizzbuzzhello",
  106,
  107,
  "fizz",
  109,
  "buzzgoodbye",
]
`;
